### PR TITLE
Add matched_queries field to SearchHit.

### LINKED ISSUE #1514: Add `matched_queries` field in SearchHit [DATAES-979]
**[Farooq Ahmad Khan](https://jira.spring.io/secure/ViewProfile.jspa?name=JIRAUSER50696)** opened **[DATAES-979](https://jira.spring.io/browse/DATAES-979?redirect=false)** and commented

`org.elasticsearch.search.SearchHit` returns a field named **`matched_queries`** but the `org.springframework.data.elasticsearch.core.SearchHit` does not hold this value. We need this value to be added in the spring data elasticsearch

---
No further details from [DATAES-979](https://jira.spring.io/browse/DATAES-979?redirect=false)

### PR DESCRIPTION
**Pull request by mdgilene**
Closes #1514

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [issue tracker](https://github.com/spring-projects/spring-data-elasticsearch/issues).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Review by sothawo — CHANGES_REQUESTED** [review]
Sorry for the delay, just have been looking at this PR and seen that I have not submitted the review, although I did the review when you submitted the PR. I could have sworn I had hit the submit button.

Two minor places: Spring Data baseline is Java 8, so there are two places that need to be changed.

**Comment by mdgilene** [comment]
Ahh Java 8 vs 9 didn't even cross my mind. Fixed!

File: src/main/java/org/springframework/data/elasticsearch/core/document/DocumentAdapters.java [hunk 1]
```diff
@@ -231,6 +233,15 @@ private static NestedMetaData from(@Nullable SearchHit.NestedIdentity nestedIden
 		return NestedMetaData.of(nestedIdentity.getField().string(), nestedIdentity.getOffset(), child);
 	}
 
+	@Nullable
+	private static List<String> from(@Nullable String[] matchedQueries) {
+		if (matchedQueries == null) {
+			return null;
+		}
+
+		return List.of(matchedQueries);
```
[the remark this answers is not in the document]
Comment (mdgilene) [reply src/main/java/org/springframework/data/elasticsearch/core/document/DocumentAdapters.java]: Fixed

File: src/test/java/org/springframework/data/elasticsearch/core/DocumentAdaptersUnitTests.java [hunk 2]
```diff
@@ -262,4 +264,18 @@ void shouldAdaptReturnedExplanations() {
...
+	void shouldAdaptReturnedMatchedQueries() {
+		SearchHit searchHit = new SearchHit(42);
+		searchHit.matchedQueries(new String[] { "query1", "query2" });
+
+		SearchDocument searchDocument = DocumentAdapters.from(searchHit);
+
+		List<String> matchedQueries = searchDocument.getMatchedQueries();
+		assertThat(matchedQueries).isNotNull();
+		assertThat(matchedQueries).hasSize(2);
+		assertThat(matchedQueries).isEqualTo(List.of("query1", "query2"));
```
[the remark this answers is not in the document]
Comment (mdgilene) [reply src/test/java/org/springframework/data/elasticsearch/core/DocumentAdaptersUnitTests.java]: Fixed

**Review by sothawo — APPROVED** [review]

**Comment by sothawo** [comment]
Thanks again for this PR.

**Pull request merged**